sentinelhub/api/batch/utils.py

```diff
@@ -50,11 +112,77 @@ def monitor_batch_job(
+    batch_kind = _auto_adjust_kind(batch_request, batch_kind)
 
-    batch_request = monitor_batch_analysis(batch_request, config, sleep_time=analysis_sleep_time)
+    batch_request = monitor_batch_analysis(batch_request, config, sleep_time=analysis_sleep_time, batch_kind=batch_kind)
     if batch_request.status is BatchRequestStatus.PROCESSING:
         LOGGER.info("Batch job is running")
 
+    if isinstance(batch_request, BatchRequest):
+        return _monitor_batch_process_execution(batch_request, config, sleep_time)
+    return _monitor_batch_statistical_execution(batch_request, config, sleep_time)  # type: ignore[func-returns-value]
+
+
+@overload
+def monitor_batch_analysis(
+    batch_request: BatchRequestSpec,
+    config: Optional[SHConfig],
+    sleep_time: int = _DEFAULT_ANALYSIS_SLEEP_TIME,
+    *,
+    batch_kind: Literal["process"],
+) -> BatchRequest:
+    pass
+
+
+@overload
+def monitor_batch_analysis(
+    batch_request: BatchRequestSpec,
+    config: Optional[SHConfig],
+    sleep_time: int = _DEFAULT_ANALYSIS_SLEEP_TIME,
+    *,
+    batch_kind: Literal["statistical"],
+) -> BatchStatisticalRequest:
+    pass
+
+
+def monitor_batch_analysis(
+    batch_request: BatchRequestSpec,
+    config: Optional[SHConfig] = None,
+    sleep_time: int = _DEFAULT_ANALYSIS_SLEEP_TIME,
+    *,
+    batch_kind: BatchKind = "process",
+) -> Union[BatchRequest, BatchStatisticalRequest]:
+    """A utility function that is waiting until analysis phase of a batch job finishes and regularly checks its status.
+    In case analysis phase failed it raises an error at the end.
+
+    :param batch_request: An object with information about a batch request. Alternatively, it could only be a batch
+        request id or a payload.
+    :param config: A configuration object with required parameters `sh_client_id`, `sh_client_secret`, and
+        `sh_auth_base_url` which is used for authentication and `sh_base_url` which defines the service deployment
+        where Batch API will be called.
+    :param sleep_time: Number of seconds between consecutive status updates during analysis phase.
+    :return: Batch request info
+    """
+    if sleep_time < _MIN_ANALYSIS_SLEEP_TIME:
+        raise ValueError(
+            f"To avoid making too many service requests please set analysis sleep time >={_MIN_ANALYSIS_SLEEP_TIME}"
+        )
+    batch_kind = _auto_adjust_kind(batch_request, batch_kind)
+
+    batch_client = _get_batch_client(batch_kind, config)
+    batch_request = batch_client.get_request(batch_request)  # type: ignore[arg-type]
+    while batch_request.status in [BatchRequestStatus.CREATED, BatchRequestStatus.ANALYSING]:
+        LOGGER.info("Batch job has a status %s, sleeping for %d seconds", batch_request.status.value, sleep_time)
+        time.sleep(sleep_time)
+        batch_request = batch_client.get_request(batch_request)  # type: ignore[arg-type]
```

Comment:
In case of a statistical batch this could actually make a call to status endpoint instead of the main endpoint. This way the size of the payload would be much smaller.

However, this would mean that we might also want that this function returns only the status payload. Because of that we might even decide to have separate functions for statistical batch even for this part.

Reply:
yes, this was also one of the reasons i was considering splitting